src/components/panel/panel.spec.js

```diff
@@ -1115,6 +1212,62 @@ describe('$mdPanel', function() {
       mdPanelPosition = $mdPanel.newPanelPosition();
     });
 
+    it('should update the position of an open panel', function() {
+      var position = null;
+
+      // Starting off with a absolute positioned panel at the top of the
+      // window.
+      testAbsTop();
```

Comment:
Each one of these should be a different test. It's hard to follow what the before and after states of each test is with this format. In tests, you want to avoid someone reviewing it having to think. I should be able to look at the test and say "Okay, there's the setup. Some action was taken. And the values changed correctly. Done."

Rarely should you need comments to explain what is happening. That's generally means the test can be better organized (functions that run asserts are generally bad).

Possibly format like this:

```
describe('should update the position of an open panel', function() {
  it('between two relative positions', function() { 
    // Create a relatively positioned panel.
    // Verify the position is initially right.
    // Call updatePosition with new relative position object.
    // Verify the position was updated.
  });

  it('between two absolute positions', function() {
    // Create an absolutely positioned panel.
    // Verify the position is initially right.
    // Call updatePosition with new absolute position object.
    // Verify the position was updated.
  });

  it('from absolute positioning to relative positioning', function() { ... });
});
```